### PR TITLE
add RACK and QUIC loss detection

### LINKED ISSUE #62: Events detected by RACK
[Yoshi said](https://mailarchive.ietf.org/arch/msg/tcpm/W-t1q8q6Q0Ofjcp0iXOgmY18mr0):
>>2: Section 3.1 "After a window reduction in response to a congestion event is
     detected by duplicate ACKs or Explicit Congestion Notification-Echo
     (ECN-Echo, ECE) ACKs [RFC3168], CUBIC remembers the congestion window..."
>
>   I think the events detected by RACK (or PTO for QUIC) can also be included here.

### PR DESCRIPTION
**Pull request by goelvidhi**
closes #62

### CODE REVIEW

File: draft-ietf-tcpm-rfc8312bis.md [hunk 1]
```diff
@@ -266,17 +266,17 @@ algorithms to AIMD TCP increase the congestion window using convex
...
-decrease of the congestion window. When CUBIC enters into congestion
-avoidance, it starts to increase the congestion window using the
-concave profile of the cubic function. The cubic function is set to
-have its plateau at the remembered congestion window size, so that the
-concave window increase continues until then. After that, the cubic
-function turns into a convex profile and the convex window increase
-begins.
+After a window reduction in response to a congestion event detected
+by duplicate ACKs, Explicit Congestion Notification-Echo (ECN-Echo,
+ECE) ACKs {{!RFC3168}}, TCP RACK {{!RFC8985}} or QUIC loss detection,
```
Comment (larseggert) [inline draft-ietf-tcpm-rfc8312bis.md:271]:
```suggestion
ECE) ACKs {{!RFC3168}}, TCP RACK {{!RFC8985}} or QUIC loss detection {{!RFC9002}},
```

File: draft-ietf-tcpm-rfc8312bis.md [hunk 2]
```diff
@@ -266,17 +266,17 @@ algorithms to AIMD TCP increase the congestion window using convex
...
+CUBIC remembers the congestion window size where it received the congestion
```
Comment (larseggert) [inline draft-ietf-tcpm-rfc8312bis.md:272]:
```suggestion
CUBIC remembers the congestion window size at which it received the congestion
```

File: draft-ietf-tcpm-rfc8312bis.md [hunk 3]
```diff
@@ -461,10 +461,9 @@ increasing the congestion window only at the reception of an ACK. It
 does not make any changes to the TCP Fast Recovery and Fast Retransmit
 algorithms {{!RFC6582}}{{!RFC6675}}.
 
-During congestion avoidance after a congestion event where a packet
-loss is detected by duplicate ACKs or by receiving packets carrying
-ECE flags {{!RFC3168}}, CUBIC changes the window increase function of
-AIMD TCP.
+During congestion avoidance after a congestion event is detected
```
Comment (larseggert) [inline draft-ietf-tcpm-rfc8312bis.md:464]:
```suggestion
During congestion avoidance, after a congestion event is detected
```

**Review by larseggert — CHANGES_REQUESTED** [review]

File: draft-ietf-tcpm-rfc8312bis.md (hunk 1 above)
Comment (goelvidhi) [reply draft-ietf-tcpm-rfc8312bis.md:271]: I had this in the first commit and I had to remove as it failed to compile. Does 9002 have a valid reference?

**Comment by goelvidhi** [comment]
@lisongxu @nsdyoshi could you review? We would like to publish a new draft on Monday

**Review by lisongxu — APPROVED** [review]
Thanks!

**Comment by lisongxu** [comment]
Done. Thank you, Vidhi!
Lisong

________________________________

File: draft-ietf-tcpm-rfc8312bis.md (hunk 1 above)
Comment (larseggert) [reply draft-ietf-tcpm-rfc8312bis.md:271]: It should? Let me try locally.

**Pull request merged**